### PR TITLE
Add feature to specify a nodeSelector for the postgreSQL instance

### PR DESCRIPTION
**Pull request by TheBigLee**
## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Update the documentation.
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

File: apis/vshn/v1/dbaas_vshn_postgresql.go [hunk 1]
```diff
@@ -64,6 +67,12 @@ type VSHNPostgreSQLServiceSpec struct {
 	PostgreSQLSettings runtime.RawExtension `json:"pgSettings,omitempty"`
 }
 
+// VSHNPostgreSQLSchedulingSpec contains settings to control the scheduling of an instance.
+type VSHNPostgreSQLSchedulingSpec struct {
```
Comment (zugao) [inline apis/vshn/v1/dbaas_vshn_postgresql.go:71]: Isn't the name a bit ambiguous? The customer may think it's PostgreSQL related.

**Review by glrf — APPROVED** [review]

**Comment by glrf** [comment]
This looks good to me. My only question is if we really want to expose the whole node selector. Only exposing the node class would probably be a better UX on APPUiO Cloud (but won't make sense once we are on APPUiO Managed so :shrug: )

**Comment by TheBigLee** [comment]
> This looks good to me. My only question is if we really want to expose the whole node selector. Only exposing the node class would probably be a better UX on APPUiO Cloud (but won't make sense once we are on APPUiO Managed so shrug )

I think it makes sense to expose the whole node selector, since the end-user has to specify the whole node selector on his pod or namespace as well. In this way the behavior is almost consistent.
The most consistent way however, would be to have a `spec.nodeSelector` in the claim and patching that to the `sgCluster` object.
What do you think?

File: apis/vshn/v1/dbaas_vshn_postgresql.go (hunk 1 above)
Comment (TheBigLee) [reply apis/vshn/v1/dbaas_vshn_postgresql.go]: Renamed it to `VSHNDBaaSSchedulingSpec`. Does that make more sense?

File: apis/vshn/v1/dbaas_vshn_postgresql.go [hunk 2]
```diff
@@ -64,6 +67,12 @@ type VSHNPostgreSQLServiceSpec struct {
 	PostgreSQLSettings runtime.RawExtension `json:"pgSettings,omitempty"`
 }
 
+// VSHNDBaaSSchedulingSpec contains settings to control the scheduling of an instance.
+type VSHNDBaaSSchedulingSpec struct {
+	// NodeSelector is a selector which must match a node’s labels for the pod to be scheduled on that node
+	NodeSelector runtime.RawExtension `json:"nodeSelector,omitempty"`
```
Comment (glrf) [inline apis/vshn/v1/dbaas_vshn_postgresql.go:73]:
```suggestion
	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
```

I didn't see that last time. I'd suggest to be more restrictive and use the same type as upstream: https://github.com/kubernetes/api/blob/master/core/v1/types.go#L3191-L3197

**Review by glrf — APPROVED** [review]

**Comment by glrf** [comment]
> I think it makes sense to expose the whole node selector, since the end-user has to specify the whole node selector on his pod or namespace as well. In this way the behavior is almost consistent.
> The most consistent way however, would be to have a `spec.nodeSelector` in the claim and patching that to the `sgCluster` object.
> What do you think?

I think it's fine as is right now. You're right it's more consistent for the user to specify the whole `nodeSelector`.

File: apis/vshn/v1/dbaas_vshn_postgresql.go (hunk 1 above)
Comment (zugao) [reply apis/vshn/v1/dbaas_vshn_postgresql.go]: yes, don't forget to update the second PR as well.

**Review by zugao — APPROVED** [review]
LGTM

**Pull request merged**